common/workunit/workunit.cpp

```diff
@@ -6260,6 +6281,15 @@ void CLocalWorkUnit::noteFileRead(IDistributedFile *file)
     }
 }
 
+void CLocalWorkUnit::noteFieldUsage(IPropertyTree * fieldUsage)
+{
+    if (fieldUsage)
+    {
+        CriticalBlock block(crit);
+        p->addPropTree("usedsources", LINK(fieldUsage));
```

Comment:
fieldUsage solely is owned by enclosing generateStatistics function, and will be deallocated as soong as we exit the function.

However, field usage is not actually copied down to WU until later point, so we need to LINK the field usage to keep it in memory (within p) and avoid premature deallocation. 

field usage will be deallocated properly when its owning tree, p, is deallocated when its owning WU is destructed.

However, I moved the LINK() out of noteFieldUsage() and put it in enclosing generateStatistics() function to semantically indicate that noteFieldUsage() function will use passed fieldUsage at later point, and need not be deallocated immediately.